pkg/agent/types/keyvalue_store.go

```diff
@@ -35,10 +33,10 @@ type Iterator interface {
 
 // Updater is an interface for bulk replacing data in a key/value store.
 type Updater interface {
-	// Update decodes data as a JSON-encoded map of key-value pairs, and inserts
+	// Update takes a sequence of alternating key-value pairs, and inserts
 	// these key-values into the store. Any preexisting keys in the store which
 	// do not exist in data will be deleted.
-	Update(data io.Reader) error
```

Comment:
why use `Update(pairs ...string)` instead of `Update(map[string]string)`? Browsing through a few of the implementations seems many convert from or to `map[string]string`

Reply:
Good catch - this was a change originally as a side effect of using generics. Now the JSON decoding is removed, and the map is more simple to use!